project/generate_dynamic_defs/dynamic_definition_builder.ts

```diff
@@ -0,0 +1,38 @@
+import { AssetSourceFile } from "../assets/asset_source_file"
+import { TsGdProject } from "../project"
+import buildActionNames from "./build_action_names"
+import buildAssetPathsType from "./build_asset_paths"
+import buildGroupTypes from "./build_group_types"
+import buildNodePathsTypeForScript from "./build_node_paths"
+import buildSceneImports from "./build_scene_imports"
+
+export default class DynamicDefinitionBuilder {
+  constructor(private project: TsGdProject) {}
+  buildActionNames() {
+    buildActionNames(this.project)
+  }
+  buildAssetPathsType() {
+    buildAssetPathsType(this.project)
+  }
+  buildGroupTypes() {
+    buildGroupTypes(this.project)
+  }
+  buildNodePathsTypeForScript(script: AssetSourceFile) {
+    buildNodePathsTypeForScript(script, this.project)
+  }
+  buildSceneImports() {
+    buildSceneImports(this.project)
+  }
+
+  public async build(scripts: AssetSourceFile[]) {
```

Comment:
naming nit: perhaps buildProject to indicate we're rebuilding everything? Additionally, I don't think this has to be async.

Reply:
Sure, can expand that.
It currently doesn't need async, but it imo is easier to preactively declare async if you ever might need async in the future (which we likely could do as it is a lot if IO).
Can change to buildProject sure.